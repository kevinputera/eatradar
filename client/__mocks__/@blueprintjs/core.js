/**
 * This manual mock only mocks Toaster, Position, and Intent component
 * and is only used in StatusMessage.test.js
 */
const core = jest.requireActual('@blueprintjs/core');

const clearMock = jest.fn();
const showMock = jest.fn();
const Toaster = {
  create: () => ({
    clear: clearMock,
    show: showMock,
  }),
};

module.exports = {
  ...core,
  clearMock,
  showMock,
  Toaster,
};
