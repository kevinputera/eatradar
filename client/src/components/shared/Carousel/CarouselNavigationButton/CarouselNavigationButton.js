import React from 'react';
import { Button } from '@blueprintjs/core';

import './CarouselNavigationButton.css';

function CarouselNavigationButton(props) {
  return (
    <div className="carousel-nav-button">
      <Button
        small
        minimal
        icon={props.icon}
        onClick={props.handleClick}
        style={{ borderRadius: '20px' }}
      />
    </div>
  );
}

export default CarouselNavigationButton;
