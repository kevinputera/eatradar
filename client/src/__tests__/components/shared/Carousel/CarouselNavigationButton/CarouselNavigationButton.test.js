import React from 'react';
import { create } from 'react-test-renderer';
import { shallow } from 'enzyme';
import { Button } from '@blueprintjs/core';

import CarouselNavigationButton from '../../../../../components/shared/Carousel/CarouselNavigationButton/CarouselNavigationButton';

describe('tests for CarouselNavigationButton component', () => {
  test('render must match snapshot', () => {
    const icon = 'chevron-left';
    const func = () => {};
    const tree = create(
      <CarouselNavigationButton icon={icon} handleClick={func} />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('button click must trigger handleClick prop', () => {
    const icon = 'chevron-left';
    const handleClick = jest.fn();
    const wrapper = shallow(
      <CarouselNavigationButton icon={icon} handleClick={handleClick} />
    );
    expect(handleClick).toHaveBeenCalledTimes(0);
    wrapper.find(Button).simulate('click');
    expect(handleClick).toHaveBeenCalledTimes(1);
  });
});
