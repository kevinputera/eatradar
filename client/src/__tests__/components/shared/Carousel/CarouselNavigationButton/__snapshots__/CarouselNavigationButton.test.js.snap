// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests for CarouselNavigationButton component render must match snapshot 1`] = `
<div
  className="carousel-nav-button"
>
  <button
    className="bp3-button bp3-minimal bp3-small"
    onClick={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    style={
      Object {
        "borderRadius": "20px",
      }
    }
    type="button"
  >
    <span
      className="bp3-icon bp3-icon-chevron-left"
      icon="chevron-left"
    >
      <svg
        data-icon="chevron-left"
        height={16}
        viewBox="0 0 16 16"
        width={16}
      >
        <desc>
          chevron-left
        </desc>
        <path
          d="M7.41 8l3.29-3.29c.19-.18.3-.43.3-.71a1.003 1.003 0 0 0-1.71-.71l-4 4C5.11 7.47 5 7.72 5 8c0 .28.11.53.29.71l4 4a1.003 1.003 0 0 0 1.42-1.42L7.41 8z"
          fillRule="evenodd"
        />
      </svg>
    </span>
  </button>
</div>
`;
